### PR TITLE
Remove the deprecated `weights_save_path` Trainer argument

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Removes 
- `Trainer(weights_save_path=...)` argument
- `Trainer.weights_save_path` property

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

cc @borda @tchaton @rohitgr7 @justusschock @akihironitta

### CODE REVIEW

File: tests/tests_pytorch/loggers/test_all.py [hunk 1]
```diff
@@ -157,64 +156,6 @@ def log_metrics(self, metrics, step):
...
-    """Test the combinations of save_dir, weights_save_path and default_root_dir."""
-
-    with contextlib.ExitStack() as stack:
-        for mgr in LOGGER_CTX_MANAGERS:
-            stack.enter_context(mgr)
-        _patch_comet_atexit(monkeypatch)
-        _test_loggers_save_dir_and_weights_save_path(tmpdir, CometLogger)
-
-
-def _test_loggers_save_dir_and_weights_save_path(tmpdir, logger_class):
```
Comment (awaelchli) [inline tests/tests_pytorch/loggers/test_all.py:171]: We could keep the test and replace all `weights_save_path` with `ModelCheckpoint(dirpath=...)`.
But then all this test does is assert that model checkpoint dirpath is correct and that it is different from loggers save_dir. Since we already have tests for each of these, there is basically no value in keeping it.

**Review by carmocca — APPROVED** [review]

**Review by rohitgr7 — APPROVED** [review]

**Review by Borda — APPROVED** [review]

**Review by Felonious-Spellfire — APPROVED** [review]
LGTM

**Review by tchaton — APPROVED** [review]
LGTM !
